source/target/freescale/target_reset_mimxrt.c

```diff
@@ -0,0 +1,54 @@
+/**
+ * @file    target_reset_mimxrt.c
+ * @brief   Target reset for the i.MX RT series
+ *
+ * DAPLink Interface Firmware
+ * Copyright (c) 2009-2016, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "target_reset.h"
+#include "swd_host.h"
+#include "info.h"
+
+
+void target_before_init_debug(void)
+{
+    swd_set_target_reset(0);
+}
+
+void prerun_target_config(void)
+{
+    target_set_state(RESET_RUN);
```

Comment:
Why does the target need to be put into the running state on boot? Is it not running already?

Reply:
Good question. I didn't do the original port, so I'll ask the author (Alex Yang). Will get back to you.

Reply:
@flit any updates on this?